mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulGameScreen.kt

```diff
@@ -74,6 +74,18 @@ class SnapshotChessBoardState(
   override var selectedPosition by mutableStateOf<ChessBoardState.Position?>(null)
     private set
 
+  override val checkPosition: ChessBoardState.Position?
+    get() {
+      val nextStep = match.game.nextStep
+      if (nextStep !is NextStep.MovePiece || !nextStep.inCheck) return null
```

Comment:
Question: Why is the `!is NextStep.MovePiece`  necessary?

Reply:
If the game's `nextStep` is a `Stalemate` for instance, we don't want to highlight the king but display some other UI instead.